components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -1118,6 +1124,35 @@ private class MonitoredItemWrapper {
                             eventFilter.AddSelectClause(ObjectTypeIds.ConditionType, "Retain");
                         }
                     }
+
+                    var sb = new StringBuilder();
+
+                    // let's loop thru the select clause and setup the field names
+                    foreach (var selectClause in eventFilter.SelectClauses) {
```

Comment:
Maybe make this an extension method so we can test this part separately?

Reply:
MonitoredItemWrapper is a private class declared inside SubscriptionServices - so it's hard to do an extension method. Let's handle this in a separate refactoring PR.